### PR TITLE
Clean up unnecessary vector constructions, and some other refactorings

### PR DESCRIPTION
**Pull request by tranzystorek-io**
Highlights:

- Use `IntoIter` for array where applicable
- Use raw strings instead of manually escaping quotes and backslashes
- Declare `Vecs` with `vec![]` (square brackets)
- Import external macros the new way: [`use crate::macro;`](https://stackoverflow.com/a/62267586/113632)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -612,7 +612,7 @@ mod cache_tests {
             }
             Ok(())
         }
-        let mut paths = vec!();
+        let mut paths = vec![];
```
Comment (dimo414) [inline src/lib.rs:615]: Why this change?

**Comment by dimo414** [comment]
This is great, thank you! Clearly I was `vec!()`-happy in my tests :)

File: src/lib.rs (hunk 1 above)
Comment (tranzystorek-io) [reply src/lib.rs:615]: For style, generally vec! uses square brackets because it is used the same as the array initializer (`[a, b, c]`)
